### PR TITLE
docs: refactor installation instructions

### PR DESCRIPTION
**Pull request by andrii-bodnar**
# Description

This pull request aims to reduce user confusion in the **Getting Started** section of the website and put all installation and basic setup instructions in one place. This also makes the **Getting Started** section more framework/toolkit-agnostic.

A quick overview of the changes:

- Added the new **_Installation and Setup_** page to the **_Getting Started_** section.
- The **_Setup with React_** (_React Project_) page has been removed in favor of  **_Installation and Setup_**.
- The **_Setup with Vite_** (_Vite Project_) page has been removed in favor of  **_Installation and Setup_**.
- The installation instructions from other places (JavaScript guide, Macros) have also been moved to the **_Installation and Setup_** page (except the instructions for individual packages like CLI, SWC Plugin, etc.).
- Other minor changes.

From the reader's perspective, the _**React Project**_ and **_Vite Project_** were confusing since it's not clear why it was differentiated into React and Vite and it could lead to wrong assumptions or questioning all of this even more.

In general, it should reduce code duplication and make reading easier for the developer.

🖼️ [Deployment Preview](https://js-lingui-git-docs-refactor-installation-crowdin.vercel.app/)

## Types of changes

[//]: # (What types of changes does your code introduce to Lingui?)
[//]: # (_Put an `x` in the boxes that apply_)

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update
- [ ] Examples update

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [x] I have read the [CONTRIBUTING](https://github.com/lingui/js-lingui/blob/main/CONTRIBUTING.md) and [CODE_OF_CONDUCT](https://github.com/lingui/js-lingui/blob/main/CODE_OF_CONDUCT.md) docs
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary documentation (if appropriate)

### CODE REVIEW

**Comment by timofei-iatsenko** [comment]
I'm reviewing thru reading docs on the deployment preview, will write down here a notes: 

- Lets remove "npm install --save-dev @babel/core" from installation for babel. I don't know why it needed. 
- Let's use the swc / babel tabs switch for Vite  `@vitejs/plugin-react` / ` @vitejs/plugin-react-swc`
- Let's not advertise 
   > .linguirc file in JSON format.
   > lingui section in package.json.

  instead, advertise only `js` or `ts` formats, for consistency. 

I'm continuing reviewing

**Comment by andrii-bodnar** [comment]
@timofei-iatsenko thanks for the quick review. Just pushed the updated version.

**Comment by andrii-bodnar** [comment]
I also think it would be nice to add Rspack and Webpack to the Build Tools section in the new article.

File: website/docs/installation.mdx [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+---
+
+import Tabs from "@theme/Tabs";
+import TabItem from "@theme/TabItem";
+
+# Installation and Setup
+
+Lingui is more than just a package; it's a comprehensive suite of tools designed to simplify internationalization. You have the flexibility to choose the specific tools that best fit your project's needs.
+
+Learn how to install Lingui in your project, whether you use JavaScript, React or React Native. Lingui also supports various transpilers and build tools, such as Babel, SWC, and Vite.
```
Comment (vonovak) [inline website/docs/installation.mdx:13]:
```suggestion
Learn how to install Lingui in your project, whether you use JavaScript, React (including RSC) or React Native. Lingui also supports various transpilers and build tools, such as Babel, SWC, and Vite.
```

File: website/docs/installation.mdx [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+## Prerequisites
+
+- Make sure you have [Node.js](https://nodejs.org/) installed (v20 or higher).
+- Install [Lingui CLI](/docs/ref/cli.md) to manage your translations and catalogs.
+
+:::tip
+Don't miss the [Lingui ESLint Plugin](/docs/ref/eslint-plugin.md) which can help you find and prevent common l10n mistakes in your code.
+:::
+
+## Choosing Transpiler
```
Comment (vonovak) [inline website/docs/installation.mdx:24]:
```suggestion
## Choosing a Transpiler
```

File: website/docs/installation.mdx [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+
+> A transpiler converts code within a language, transforming newer features into older equivalents for compatibility, or expanding concise syntax into more detailed implementations.
```
Comment (vonovak) [inline website/docs/installation.mdx:26]:
```suggestion
> A transpiler converts code within a language, transforming newer features into older equivalents for compatibility, or expanding concise syntax into more verbose implementations.
```

File: website/docs/installation.mdx [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+
+Lingui needs a transpiler to work. It's responsible for transforming Lingui's JS/JSX components into [ICU Message Format](/docs/guides/message-format.md) and extracting message IDs. Both Babel and SWC transpilers are supported. Depending on your project setup, you can choose the one that suits your needs.
```
Comment (vonovak) [inline website/docs/installation.mdx:28]:
```suggestion
Lingui needs a transpiler to work. It's responsible for transforming Lingui's JS/JSX components into [ICU Message Format](/docs/guides/message-format.md) and extracting message IDs. Both Babel and SWC transpilers are supported. Follow the Babel or SWC setup depending on what transpiler your project already uses.
```

File: website/docs/ref/conf.md [hunk 5]
```diff
@@ -1,47 +1,21 @@
...
-  "locales": [],
-  "extractors": ["babel"],
-  "orderBy": "messageId",
-  "pseudoLocale": "",
-  "rootDir": ".",
-  "runtimeConfigModule": ["@lingui/core", "i18n"],
-  "sourceLocale": ""
-}
-```
+In the case of TypeScript based config you can use ESM format and _export default_.
```
Comment (vonovak) [inline website/docs/ref/conf.md:18]:
```suggestion
In the case of TypeScript-based config you can use ESM format and _export default_.
```

**Comment by andrii-bodnar** [comment]
@vonovak thanks for the review! I have updated everything according to your suggestions.

File: website/docs/ref/conf.md (hunk 5 above)
Comment (timofei-iatsenko) [reply website/docs/ref/conf.md]: I think this statement is not full, i believe the config will follow the `type: module` in the closes package json. So if the user uses ESM for the project he also could use ESM for the config. But i didn't test it though. TBA

Comment (andrii-bodnar) [reply website/docs/ref/conf.md]: Probably related to this - https://github.com/lingui/js-lingui/issues/1704

**Pull request merged**